_posts/2020-01-10-ios-logout-button-at-last.markdown

```diff
@@ -0,0 +1,126 @@
+---
+layout: epic
+title: "At Last: A Log Out Button"
+date: "2020-01-14"
+author: [ash]
+categories: [ios, tech debt, hackathon]
+---
+
+In 2013, Artsy shipped the first version of our iOS app. Typical for an early-day startup, the app was a "minimum
+viable product" (with a big emphasis on "minimum"). One of the features that didn't make the cut was something you
+expect to see in most apps: a log out button.
+
+When I joined Artsy a year later, there was still no log out button. And there would be no log out button for
+another six years, until today.
+
+I want to talk about this quirk of our app, from both product and technical perspectives. Why wasn't this already
+in our app? Why was it so difficult to build? These are interesting questions, and their answers shed light on how
+products mature over time. I also want to talk about how we finally managed to prioritize this kinda weird feature
+request (spoilers: it was our company-wide hackathon). Let's go!
+
+<!-- more -->
+
+When I say that our app doesn't have a log out button, that's a bit of a fib: it _does_ have a log out button... in
+the admin-only debug menu. The reason why this isn't a user-facing feature is that the final step of this
+admin-only log out feature is a call to `exit(0)`, effectively crashing the app. That's _one_ way to make sure that
+user-specific state doesn't pollute your app's runtime, but of course it would be ridiculous for a user-facing log
+out button to crash the app. The only other way to log out was to uninstall the app, which is _not_ something we
+want to encourage users to do.
+
+So Artsy staff could log out of the app, but our normal users couldn't. This quirk was acceptable in the early days
+of our app, but as the years wore on, it became less of a quirk and more of a product limitation. Even three months
+after creating our new Mobile Experience team, we hadn't yet prioritized this feature. Let's
+[dive into this question](https://en.wikipedia.org/wiki/Five_whys) a little bit.
+
+The log out button hadn't been prioritized. **Why?** Well, to be honest, we had – and still have,
+[we're hiring by the way](http://artsy.net/jobs) – a lot of work to do to improve the app. Among the high-impact
```

Comment:
Think "we're hiring by  the way" might work better at end of this sentence in parentheses maybe?